src/cpp/fil.cpp

```diff
@@ -8,18 +26,26 @@ adapted from samtools/calDep.c
 #include <gsl/gsl_sf.h>
 #include <cstdio>
 #include <fstream>
+#include <sstream>
 #include <iostream>
 #include <map>
 #include <string>
-#include "faidx.h"
-#include "sam.h"
+#include <functional>
+#include <limits>
+
+#include <string.h>
+#include <assert.h>
```

Comment:
Also, this is C++, so use the `<cXXX>` instead

Reply:
Well, in my opinion [the jury is still out](https://stackoverflow.com/a/8380835/16168) which practice is better.

Basically, `xxx.h` guarantees you that the functions will be in the global name space (and they might appear in the `std::` name space too) and `cxxx` guarantees you that the functions will be int he standart name space (And they might appear in the global one, as is the case with gnu).

Also, `cxxx` can provide multiple signatures and overload functions.
While it makes definitely sense for some functions (`std::abs` will be defined separately for floating-point types and integers, whereas `abs` might be integer-only and corrupt floating point inputs), in our case the point is moot as memory and string functions always work on the same pointer types and they will get replaced by intrinsic inlines and get optimized according to surrounding code so constness won't matter much in practice.

But okay, I'll replace the `cxxx` headers and `std::` prefixes